src/main/java/org/zalando/problem/spring/web/advice/validation/ConstraintViolationProblem.java

```diff
@@ -38,16 +39,16 @@
     public static final URI CONSTRAINT_VIOLATION = URI.create(CONSTRAINT_VIOLATION_VALUE);
 
     private final Optional<String> detail;
-    private final ImmutableList<Violation> violations;
+    private final List<Violation> violations;
 
-    public ConstraintViolationProblem(final ImmutableList<Violation> violations) {
+    public ConstraintViolationProblem(final List<Violation> violations) {
```

Comment:
Using this constructor I could still create a mutable instance. Needs to do another copy, or maybe the sorting logic could be moved here instead of being in the advise.